Objective-C/CBLDatabase.mm

```diff
@@ -327,6 +334,35 @@ - (BOOL) purgeDocumentWithID: (NSString*)documentID error: (NSError**)error {
     }
 }
 
+#pragma mark - Blob Save/Get
+
+- (BOOL) saveBlob: (CBLBlob*)blob error: (NSError**)error {
+    return [blob installInDatabase: self error: error];
+}
+
+- (CBLBlob*) getBlob: (NSDictionary*)dict {
+    if (!dict[kBlobDigestProperty] || !dict[kBlobContentTypeProperty])
+        [NSException raise: NSInvalidArgumentException
+                    format: @"Property dictionary is missing the digest or content-type"];
+    
+    // Read blob from the BlobStore:
+    C4BlobStore* blobStore = [self getBlobStore: nullptr];
+    C4BlobKey key;
+    
+    if(!c4blob_keyFromString(CBLStringBytes(asString(dict[kBlobDigestProperty])), &key))
+        return nil;
+    
+    //TODO: If data is large, can get the file path & memory-map it
+    FLSliceResult res = c4blob_getContents(blobStore, key, nullptr);
+    NSData* content = sliceResult2data(res);
+    FLSliceResult_Release(res);
+    
+    NSMutableDictionary *tempProps = [dict mutableCopy];
+    tempProps[kBlobDataProperty] = content;
+    tempProps[kBlobLengthProperty] = @(content.length);
+    return [[CBLBlob alloc] initWithDatabase: self properties: tempProps];
```

Comment:
What needs to be passed into the properties is the given dict from the method's parameter.

Reply:
ok. this means, 
- here we will only pass the db instance and the props to Blob object. 
- inside the blob class constructor, we will get the content (move the logic to blob class)
- populates the content, content-length, content-type etc from blob constructor

Reply:
You can also make the init to the assertion to check the required properties (not sure if this will affect any existing functionalities or not).

Reply:
assertion check will fail other tests, since the same function without the digest information also uses the same constructor. So not going to add the assertion check.